AEPCampaignClassic/Sources/CampaignClassicRegistrationManager.swift

```diff
@@ -0,0 +1,144 @@
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import AEPServices
+import AEPCore
+
+///
+/// Class that manages Campaign Classic device registration requests.
+///
+class CampaignClassicRegistrationManager {
+
+    private var runtime: ExtensionRuntime
+    private var systemInfoService: SystemInfoService {
+        ServiceProvider.shared.systemInfoService
+    }
+
+    /// DataStore instance to access Campaign Classic's stored properties.
+    let datastore = NamedCollectionDataStore(name: CampaignClassicConstants.EXTENSION_NAME)
+
+    /// A variable holding the SHA256 hashed device registration information.
+    /// The following variable is a hash of deviceToken, userKey and xml serialized additional parameters provided from the user.
+    /// This property is set if the device is successfully registered with the campaign servers.
+    /// This property is then used to hold back future device registration calls with unchanged data.
+    /// The setter and getter of this property will use the datastore to store and retrieve the information.
+    private(set) var hashedRegistrationData: String? {
+        get {
+            datastore.getString(key: CampaignClassicConstants.DatastoreKeys.TOKEN_HASH)
+        }
+        set {
+            datastore.set(key: CampaignClassicConstants.DatastoreKeys.TOKEN_HASH, value: newValue)
+        }
+    }
+
+    /// Initializer for CampaignClassicRegistrationManager with extensionRuntime instance.
+    /// Extension runtime is used for retrieving the configuration shared state.
+    /// - Parameter runtime: the Campaign Classic extension's runtime
+    init(_ runtime: ExtensionRuntime) {
+        self.runtime = runtime
+    }
+
+    /// Sends a device registration request to the configured Campaign Classic server.
+    /// If configuration is not available or Campaign Classic marketing server is not configured, no request is sent.
+    /// If registration information has not changed since the last request, no request is sent.
+    /// If the privacy is opted out , no request is sent.
+    /// - Parameter event: the campaign registration request event containing all the device/user details
+    func registerDevice(event: Event) {
+
+        /// retrieve the device token from the event
+        /// device token is the unique token received from Apple push notification service through the application
+        /// bail out from the registration request if device token is unavailable
+        guard let deviceToken = event.deviceToken else {
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration failed, device token is not available.")
+            return
+        }
+
+        /// bail out if the privacy is opted out.
+        let configuration = CampaignClassicConfiguration.init(forEvent: event, runtime: runtime)
+        guard configuration.privacyStatus == PrivacyStatus.optedIn else {
```

Comment:
how do we handle unknown privacy status for ACC?  should this check be `.privacyStatus != .optUnkown` to match the comment?

Reply:
Privacyunknown status will not send out registration requests as well. I should change the comment here